src/cms/utils/filter_links.py

```diff
@@ -1,33 +1,78 @@
-from django.db.models import Q
+from cms.models import PageTranslation, EventTranslation, POITranslation
 
-from linkcheck.models import Link
 
-
-def filter_links(region_slug):
+def filter_links(region_slug, link_filter=None):
     """
-    First filters links by current region
-    Then filters by the link state
-    Finally returns resulting query sets
+    Filter all links of one region by the given category
 
     :param region_slug: the current regions slug
     :type region_slug: str
-    :return: dictionary of filtered querysets
-    :rtype: dict
+
+    :param link_filter: which links should be returned (one of ``valid``, ``invalid``, ``ignored``, ``unchecked``).
+                        If parameter is not in these choices or omitted, all links are returned by default.
+    :type link_filter: str
+
+    :return: a tuple of the requested links and a dict containing the counters of all remaining links
+    :rtype: tuple
     """
-    qset = Link.objects.filter(
-        Q(page_translations__page__region__slug=region_slug)
-        | Q(event_translations__event__region__slug=region_slug)
-        | Q(poi_translations__poi__region__slug=region_slug)
+    # Get all translation objects of this region and prefetch its links
+    page_translations = list(
+        PageTranslation.objects.filter(page__region__slug=region_slug)
+        .distinct("page")
+        .select_related("language")
+        .prefetch_related("links__url")
+    )
```

Comment:
It might make sense to also fetch `content_type` since that is accessed in `link_list_row.html`, too.
It looks like this change reduces the number of queries by one per link.

```suggestion
        .prefetch_related("links__content_type")
    )
```

Reply:
Ah, very good idea! :+1:

Although, now that I think about it: probably we can circumvent the query altogether by getting the type from the `item.translation` object directly via `item.translation._meta.verbose_name`. The only problem is that private members cannot be referenced in the template, so maybe we have to add another templatetag to return the readable type from a model instance?

```
@register.simple_tag
def get_model_verbose_name(instance):
    return instance._meta.verbose_name
```
and then in `link_list_row.html`:
```
<td class="pr-2">{% get_model_verbose_name item.translation %}</td>
```

Reply:
I tried to address the sql queries of `content_type`, but somehow they still appear in the Django debug toolbar... It seems that `prefetch_related("links__url")` implicitly also queries for the `content_type`, even if it's unused in the template. Do you have another idea what might cause this? If not, no problem, since this is only one query per content type indepdendently from the number of links in the database, I don't this this is causing too much overhead.

Reply:
I cannot find an sql query that queries for `content_type`. Can you maybe copy the query so I can take a look at it?